### ISSUE TITLE
Load failed - possible bug in OBO reading code

### ISSUE THREAD

**Issue by kimrutherford** [pombase/pombase-chado#1178]
Here is the error.  It fails while loading GO but it looks like a bad interaction with MONDO.

-------------------------------------------------

reading: go-basic.obo
failed - rolling back: failed while reading go-basic.obo: two previously read terms match an alt_id field from:
[Term]
id: GO:0000781
name: chromosome, telomeric region
alt_id: GO:0000784
comment: Note that this term can be used in place of the obsolete cellular component term 'telomere ; GO:0005696'. Use with caution because this term refers to a specific region of the chromosome and not a protein complex.                                                 
def: The end of a linear chromosome, required for the integrity and maintenance of the end. A chromosome telomere usually includes a region of telomerase-encoded repeats the length of which rarely exceeds 20 bp each and that permits the formation of a telomeric loop (T-loop). The telomeric repeat region is usually preceded by a sub-telomeric region that is gene-poor but rich in repetitive elements. Some telomeres only consist of the latter part (for eg. D. melanogaster telomeres). [GOC:elh]                                                
namespace: cellular_component
relationship: is_a GO:0098687
synonym: nuclear chromosome, telomere NARROW []
synonym: nuclear chromosome, telomeric region NARROW []
synonym: telomere RELATED []
xref: SO:0000624

matching term 1:
[Term]
id: GO:0000781
name: chromosome, telomeric region
namespace: mondo
relationship: is_a GO:0098687

matching term 2:
[Term]
id: GO:0000784
is_obsolete: 1
namespace: mondo
replaced_by: GO:0000781

**Comment by ValWood** [comment]
I didn't know there were ontology  links between GO and Mondo.....

**Comment by kimrutherford** [comment]
There was a bug (kind-of) in the OBO loading code.  There are lots of edge cases to handle and it was missing the special case where an alt_id is also the ID of an obsolete term.  This normally shouldn't happen but it seems to be because the GO terms in the MONDO OBO file are from an older release than the GO terms we get from go-basic.obo.  I suspect that the problem would go away in the next MONDO release.

But I've changed the OBO loading code to cope gracefully with this weird situation.

I've also switched from mondo.obo to mondo-simple.obo which has a lot less in the way of exciting non-MONDO terms and relationships.  This is an experiment so I'll check for problems before releasing tomorrow.

> I didn't know there were ontology links between GO and Mondo.....

GO is used for logical definitions in MONDO like:
```
relationship: disease_disrupts GO:0015349 ! thyroid hormone transmembrane transporter activity
```

mondo-simple.obo doesn't have the logical definitions so there should be less drama with it.
